File: TrussAnalysis.py
#All required info
bridge_span = 100
bridge_height = 5
percent_load = 1
train_length = 100
wheel_diatribution_train = [5,45,55,95]
force_from_wheel = 50




num_panel_points = bridge_span // bridge_height
span_panel_points = bridge_span / num_panel_points

#finds pins affected by live load
def find_pins (wheel_distribution_train, percent_load, bridge_span, train_length): 
    
    affected_pins = []
    amount_affected = []
    wheel_location = [] 
    
    #where front of train is
    front_train = (1-percent_load) * bridge_span
    
    #number of wheels is equal to number of distributed loads
    for i in range(len(wheel_distribution_train)):
        
        #checking if wheel is actually in the bridge
        if (front_train + wheel_distribution_train[i]) <= bridge_span:
            wheel_location.append(front_train + wheel_distribution_train[i])
            
            affected_pins.append(int(((wheel_location[i]/bridge_span) * num_panel_points)))
            affected_pins.append(int(((wheel_location[i]/bridge_span) * num_panel_points)) + 1)
            
            pin_affected_amount = 1 - (((wheel_location[i]/bridge_span) * num_panel_points)-int(((wheel_location[i]/bridge_span) * num_panel_points)))
            amount_affected.append(pin_affected_amount)
            amount_affected.append(1 -pin_affected_amount)
    
    
    print (amount_affected)
    return affected_pins

print(find_pins(wheel_diatribution_train, percent_load, bridge_span, train_length))


        

File: src/TrussAnalysis.java
import java.util.ArrayList;
import java.math.*;

public class TrussAnalysis {

    static Bridge bridge = new Bridge();
    static Train train = new Train();

    public static void main (String[] args) {
        double bridge_span = 100;
        double bridge_height = 5;
        int percent_load = 1;
        double train_length = 100;
        ArrayList<Double> wheelDistribution = new ArrayList<Double>();
        double force_from_wheel = 50;

        wheelDistribution.add(5.0);
        wheelDistribution.add(45.0);
        wheelDistribution.add(53.0);
        wheelDistribution.add(95.0);

        int numPanelPoints = (int) (bridge_span /bridge_height);
        double spanPanelPoints = bridge_span / numPanelPoints;

        bridge.setPanelPoints(numPanelPoints);
        bridge.setPanelPointSpan(spanPanelPoints);
        bridge.setHeight(bridge_height);
        bridge.setLength(bridge_span);

        train.setLength(train_length);
        train.setForceFromWheel(force_from_wheel);

        for (int i = 0; i < (2*numPanelPoints) - 2; i++) {
            Pin pin = new Pin();
            bridge.getBridgePins().add(pin);
        }

        findPins (wheelDistribution, percent_load);

    }

    //finds pins affected by live load
    public static void findPins (ArrayList<Double> wheelDistribution, int percent_load) {

        int jointAffected = 0;
        double amountAffected = 0;
        double forceRatio;

        double frontTrain = (1-percent_load) * bridge.getLength();
        for (int i = 0; i < wheelDistribution.size(); i++) {
            if (wheelDistribution.get(i) + frontTrain <= bridge.getLength()) {

                bridge.addWheelLocation(frontTrain + wheelDistribution.get(i));

                jointAffected = (int) (bridge.getWheelLocation().get(i)/bridge.getLength() * bridge.getPanelPoints());

                forceRatio = 1.0 - (((bridge.getWheelLocation().get(i)/bridge.getLength()) * bridge.getPanelPoints())
                        - Math.floor((((bridge.getWheelLocation().get(i)/bridge.getLength()) * bridge.getPanelPoints()))));


                amountAffected = forceRatio * train.getForceFromWheel();

                bridge.getBridgePins().get(jointAffected - 1).addForceOnPins(amountAffected);
               System.out.println(bridge.getBridgePins().get(jointAffected - 1).getForceOnPins());

               if (forceRatio < 1) {
                   amountAffected = forceRatio * train.getForceFromWheel();
                   bridge.getBridgePins().get(jointAffected).addForceOnPins(amountAffected);
               }

            }

        }

    }
}
